src/models/room.ts

```diff
@@ -2256,6 +2256,13 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
             client: this.client,
             pendingEventOrdering: this.opts.pendingEventOrdering,
             receipts: this.cachedThreadReadReceipts.get(threadId) ?? [],
+
+            // This is necessary to be able to jump to events in threads:
+            // If we jump to an event in a thread where neither the event, nor
+            // the root, nor any thread event are loaded yet, we'll load the
+            // event as well as the thread root, create the thread, and pass
+            // the event through this.
+            events,
```

Comment:
adding `events` here (to `Thread#replayEvents`) means that `Thread.constructor` inserts these correctly regardless of async ordering; previously:
1. `addLiveEvents` called with multiple thread events
2. `Thread.constructor` is called using the thread root, starts async work...
3. the original `addLiveEvents` continues and tries to add the thread responses to the newly-created (but not fully initialized) `Thread`; fails check in `Thread#addEvent` where `this.initialEventsFetched`, so fails to add events to timeline
4. continuing `Thread.constructor` async: `updateThreadMetadata` is called, and tries to get initial fetch of events. _somehow this doesn't work – I forget now how (sorry, writing these notes a few days after investigating...)_